sdk/dotnet/Helm/ChartBase.cs

```diff
@@ -152,9 +145,19 @@ private void Fetch(string chart, ChartFetchArgsUnwrap opts)
                 flags.Add("--untar");
             }
 
-            // Fallback to using the HELM_HOME environment variable if opts.home is not set.
-            if (string.IsNullOrEmpty(opts.Home)) {
-                opts.Home = Environment.GetEnvironmentVariable("HELM_HOME");
+            var env = (Dictionary<string, string>)Environment.GetEnvironmentVariables();
+
+            // Helm v3 removed the `--home` flag, so we must use an env var instead.
+            if (!string.IsNullOrEmpty(opts.Home))
+            {
+                if (env.ContainsKey("HELM_HOME"))
```

Comment:
You don't need this `if` here: the assignment style `env["HELM_HOME"] = opts.Home;` should work for both cases.